change langchain quickstart (#3531)
langchain-ai/docs: src/oss/langchain/philosophy.mdx

- It is still very early on in that transformation.
- While it's easy to build a prototype of those agentic applications, it's still really hard to build agents that are reliable enough to put into production.

> Today developers can choose how they build agents: [LangChain](/oss/langchain/overview) for flexibility and control, or [Deep Agents](/oss/langchain/overview) for a batteries-included experience with built-in planning, filesystem tools, subagents, and context management. Both are built on [LangGraph](/oss/langgraph/overview).

Reviewer: makes it sound like an "or" but deepagents also let's you do these things

Author: ```suggestion
Today developers can choose how they build agents: use [LangChain](/oss/langchain/overview) for maximum flexibility and control, or [Deep Agents](/oss/langchain/overview) which allows for similar flexibility and control but comes with opinionated built-in planning, filesystem tools, subagents, and context management. Both are built on [LangGraph](/oss/langgraph/overview).
```

    2. A standard message content format: Model APIs evolved from returning messages with a simple content string to more complex output types - reasoning blocks, citations, server-side tool calls, etc. LangChain evolved its message formats to standardize these across providers.
</Update>

> <Update label="2026-03-15">

Reviewer 2: This date is incorrect;

0.0.1 was published Jul 29, 2025

Author: I based this on when we announced it to the world. Though it's also possible we don't want this here at all until v1.0. @vtrivedy what do you think?

Reviewer: i think fine to remove date

Author: @vtrivedy that's not an option, the page uses dates in this timeline.

Author: we can
(1) not list it here
(2) list as is
(3) list with the July 29 date

Author: I'll put 0.5.3 with it, I think that makes most sense

> <Update label="2026-03-15">

Author: ```suggestion
<Update label="2026-03-15" description="v0.5.3">
```

    **Deep Agents is released** as an open-source agent harness built on LangGraph.

>     While LangChain provides flexible building blocks for custom agent architectures, [Deep Agents](/oss/langchain/overview) offers a batteries-included option for complex, long-running tasks like research and coding. It adds built-in planning tools, a virtual filesystem with pluggable backends (in-memory, disk, LangGraph store, sandboxes), subagent spawning for context isolation, and a terminal CLI. Use Deep Agents for more autonomous agents with predefined tools; use LangChain for full control over your agent architecture.

Reviewer 2: Not sure we should mention CLI

>     While LangChain provides flexible building blocks for custom agent architectures, [Deep Agents](/oss/langchain/overview) offers a batteries-included option for complex, long-running tasks like research and coding. It adds built-in planning tools, a virtual filesystem with pluggable backends (in-memory, disk, LangGraph store, sandboxes), subagent spawning for context isolation, and a terminal CLI. Use Deep Agents for more autonomous agents with predefined tools; use LangChain for full control over your agent architecture.

Author: ```suggestion
    While LangChain provides flexible building blocks for custom agent architectures, [Deep Agents](/oss/langchain/overview) offers a batteries-included option for complex, long-running tasks like research and coding. It adds built-in planning tools, a virtual filesystem with pluggable backends (in-memory, disk, LangGraph store, sandboxes), and subagent spawning for context isolation. Use Deep Agents for more autonomous agents with predefined tools; use LangChain for full control over your agent architecture.
```
